src/main/kotlin/blaster707/contactlist/Application.kt

```diff
@@ -100,7 +100,8 @@ object Application {
 
     fun handleDeleteContactEntry(req: Request): Response{
         val contactToDelete = req.bodyString()
-        return if (contactList.deleteContactEntry(contactToDelete)) {
+        return if (contactList.contactEntryIdLocated(contactToDelete)) {
+            contactList.deleteContactEntry(contactToDelete)
             Response(Status.OK).body("Contact Entry with ID $contactToDelete will be deleted on next save.")
         } else Response(Status.NOT_FOUND).body("No Contact Entry with ID $contactToDelete located.")

```

Comment:
Nice to see you using `Status.NOT_FOUND` here - we never covered all the different status codes. Good intuition. If you are curious about all the different status codes, [httpstatuses.com](http://httpstatuses.com) is a useful site.  You can also look at [this site](https://www.loggly.com/blog/http-status-code-diagram/) for a decision tree that can give some guidance on how to pick status codes. NOTE: everyone has their own opinions on HOW to pick status codes. Which rules you pick, for the most part, is not the most important thing - applying what ever rules you follow consistently is more important.